java/jdbc-reactive-extensions-intro/src/main/java/com/oracle/jdbc/reactive/SQLStatementWithAsynchronousJDBC.java

```diff
@@ -0,0 +1,126 @@
+/*
+  Copyright (c) 2024, Oracle and/or its affiliates.
+
+  This software is dual-licensed to you under the Universal Permissive License
+  (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License
+  2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose
+  either license.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+     https://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+package com.oracle.jdbc.reactive;
+
+import java.sql.SQLException;
+import java.util.concurrent.Flow;
+
+import oracle.jdbc.OracleConnection;
+import oracle.jdbc.OraclePreparedStatement;
+import oracle.jdbc.pool.OracleDataSource;
+
+public class SQLStatementWithAsynchronousJDBC {
+
+	private OracleDataSource ods = null;
+	private OracleConnection conn = null;
+
+	public SQLStatementWithAsynchronousJDBC() {
+		try {
+			ods = new OracleDataSource();
+			// jdbc:oracle:thin@[hostname]:[port]/[DB service/name]
+			ods.setURL("jdbc:oracle:thin@[hostname]:[port]/[DB service/name");
+			ods.setUser("[Username]");
+			ods.setPassword("[Password]");
+
+			conn = (OracleConnection) ods.getConnection();
+		} catch (SQLException e) {
+			e.printStackTrace();
+		}
+	}
+
+	public static void main(String[] args) {
+		SQLStatementWithAsynchronousJDBC asyncSQL = new SQLStatementWithAsynchronousJDBC();
+		try {
+			// Execute a SQL DDL statement to create a database table
+			// asynchronously
+			asyncSQL.createTable(asyncSQL.getConn());
+		} catch (SQLException e) {
+			e.printStackTrace();
+		}
+	}
+
+	/**
+	 * Asynchronously creates a new table by executing a DDL SQL statement
+	 * 
+	 * @param connection Connection to a database where the table is created
+	 * @return A Publisher that emits the result of executing DDL SQL
+	 * @throws SQLException If a database access error occurs before the DDL SQL can
+	 *                      be executed
+	 */
+	private Flow.Publisher<Boolean> createTable(OracleConnection connection) throws SQLException {
+
+		OraclePreparedStatement createTableStatement = (OraclePreparedStatement) connection
+				.prepareStatement("CREATE TABLE employee_names (" + "id NUMBER PRIMARY KEY, "
```

Comment:
Using the async APIs for a DDL wouldn't it be common. A better code sample should show a SELECT query and a DML.

Reply:
As we discussed, I don't disagree with you on that, but at the end of the day, the goal is to present the use of our async API and the extensions offering, and it can be used with any SQL statement (DDL, DQL, DML, DCL and TCL) as it will be issued in an async way regardless of the semantics of the SQL statement, so the purpose of it is not defeated at all.
I proposed to create part 2 of this blog post (which just wanted to serve as an intro as no blog post existed about such API) as a more complete segway and then explore CRUD scenarios in it, so I added it as a TODO item for me.

Reply:
Also, please note that the official documentation also uses DDL as an example - https://docs.oracle.com/en/database/oracle/oracle-database/21/jjdbc/jdbc-reactive-extensions.html#GUID-1C40C43B-3823-4848-8B5A-D2F97A82F79B